### DISCUSSION TITLE
Layout causing "Component is missing template or render function." Bug?

### DISCUSSION THREAD

**Discussion by nurdism** [nuxt/nuxt#19461] [Questions] [answered]
Am I using the layouts wrong or is this a bug? 

Simple set up, I have a layout "test" and some pages: 

`pages/test/[id]/index.vue`
`pages/test/[id].vue`

on page `pages/test/[id].vue` I have 

```vue
<script setup>
definePageMeta({
  layout: 'test',
  key: (route) => route.fullPath,
});
</script>
```

When you visit `http://foo.com/test/123132` it will not render `pages/test/[id]/index.vue` and will throw a "Component is missing template or render function."

Here is a reproduction: https://stackblitz.com/edit/github-yxafqp?file=pages/test/[id].vue 

Edit: 

if I add 

```vue
<template>
  <NuxtPage :page-key="(route) => route.fullPath" />
</template>
```

to the vue file, it works as indented, but according to the https://nuxt.com/docs/api/components/nuxt-page docs I *should* be about to just use page meta?

**Comment by TheAlexLichter** [comment, marked as answer]
You need to add a template to your page component in any case. For nested routes, you want to add `<NuxtPage>` (and whatever you want around the nested content, if any.
If you define the page key via page meta as you did above, you can omit the `pageKey` prop ☺️ 

PR welcome if that is unclear in the docs 👍

**Reply by TheAlexLichter** [reply]
The two options as examples:


## Page key via prop


Parent
```vue
<template>
  <div>
    <h1>I am the parent view</h1>
    <NuxtPage :page-key="(route) => route.fullPath" />
  </div>
</template>
```

Child
```vue
<template>
  <h1>Child Content</h1>
</template>
```` 

---

## Page key via meta

Parent
```vue
<template>
  <div>
    <h1>I am the parent view</h1>
    <NuxtPage />
  </div>
</template>
```

Child
```vue
<script setup>
definePageMeta({
  key: route => route.fullPath
})
</script>

<template>
  <h1>Child Content</h1>
</template>
````
